src/collection-view.js

```diff
@@ -295,7 +296,20 @@ Marionette.CollectionView = Marionette.AbstractView.extend({
       });
     }
 
-    return childView;
```

Comment:
>This could be problematic actually. I remember us trying something like this earlier and 
>running into bugs with identifying constructor functions.

Sorry, I'm not following.

The checks in line 300 and 304 are enough to identify if `childView` is a view constructor function (or "view class", I don't know the term that is usually used in Marionette), or just a regular function (which we assume that returns a constructor function).